vars/kibanaPipeline.groovy

```diff
@@ -311,11 +311,34 @@ def buildOss(maxWorkers = '') {
   }
 }
 
-def buildXpack(maxWorkers = '') {
+def getBuildArtifactBucket() {
+  def dir = env.ghprbPullId ? "pr-${env.ghprbPullId}" : buildState.get('checkoutInfo').branch.replace("/", "__")
+  return "gs://ci-artifacts.kibana.dev/default-build/${dir}/${buildState.get('checkoutInfo').commit}"
+}
+
+def buildXpack(maxWorkers = '', uploadArtifacts = false) {
   notifyOnError {
     withEnv(["KBN_OPTIMIZER_MAX_WORKERS=${maxWorkers}"]) {
       runbld("./test/scripts/jenkins_xpack_build_kibana.sh", "Build X-Pack Kibana")
     }
+
+    withGcpServiceAccount.fromVaultSecret('secret/kibana-issues/dev/ci-artifacts-key', 'value') {
```

Comment:
This should only happen when `uploadArtifacts` is `true` right?

Reply:
Thanks, was too distracted on this one